shared/features/filters/period-filter/period-filter.tsx

```diff
@@ -0,0 +1,98 @@
+import { Calendar } from "lucide-react";
+import { useEffect, useMemo, useState } from "react";
+
+import { bootstrap } from "@/core/bootstrap";
+import { CustomPeriodType } from "@/core/kernel/date/date-facade-port";
+
+import { Button } from "@/design-system/atoms/button/variants/button-default";
+import { DateRangePicker, DateRangePickerValue } from "@/design-system/atoms/date-range-picker";
+import { Popover } from "@/design-system/atoms/popover";
+import { Typo } from "@/design-system/atoms/typo";
+import { Menu } from "@/design-system/molecules/menu";
+
+import { PeriodFilterProps } from "@/shared/features/filters/period-filter/period-filter.types";
+import { usePeriodSelectOptions } from "@/shared/hooks/select/use-period-select-options";
+import { Translate } from "@/shared/translation/components/translate/translate";
+
+const FUTURE_DEFAULT_DATE = new Date();
+FUTURE_DEFAULT_DATE.setDate(new Date().getDate() + 30);
```

Comment:
`getDate` returns an int between 1 and 31, if you add 30 I'm not sure the month will roll over ?

Reply:
right changed to 20